### PR TITLE
MNT: Cleanup docstring substitution mechanisms

### PR DESCRIPTION
**Pull request by timhoffm**
Inspired by #28746.

Since the module `_docstring` is internal, we can do this refactoring without deprecations.

<s>It's probably easiest for review to go through the commits one after the other. Each is a simple change, but they often touch the same code so that separate PRs would depend on each other.</s>

I've stripped this down for easier review. This is now two commits, which I suggest should be kept (i.e. please do not squash)

- MNT: Make `Substitution` and `_ArtistPropertiesSubstitution independent`
  They do not have relevant functional overlap. Thus, it's simpler to keep them completely separated.
- Rename `_ArtistPropertiesSubstitution.update()` to `register()`
  While it's internally a dict.update. The logical process is a registration to make the names publically available. Also,
  remove the possibility to pass a dict. Passing kwargs is enough and we can simplify the API to only one type of usage.

### CODE REVIEW

**Comment by timhoffm** [comment]
Failing test seems unrelated (an OSX test that is marked as flaky anyway).

One could still rename `interpd` to something better. But I don't have a good name at hand right now and this PR is already large enough. I therefore suggest to defer further possible improvements to a separate PR.

File: lib/matplotlib/_docstring.pyi [hunk 1]
```diff
@@ -21,11 +21,13 @@ class _ArtistKwdocLoader(dict[str, str]):
...
 
 
-class _ArtistPropertiesSubstitution(Substitution):
+class _ArtistPropertiesSubstitution:
     def __init__(self) -> None: ...
+    def register(self, **kwargs) -> None: ...
     def __call__(self, obj: _T) -> _T: ...
 
 
+
```
Comment (QuLogic) [inline lib/matplotlib/_docstring.pyi:30]: Unnecessary change.
```suggestion
```

**Review by QuLogic — APPROVED** [review]
